scripts/ci/check_compliance.py

```diff
@@ -25,8 +25,15 @@
 sys.path.insert(0, str(Path(__file__).resolve().parents[1]))
 from get_maintainer import Maintainers, MaintainersError
 
+HERE = Path(__file__).parent
+
 logger = None
 
+# This ends up using the current repository if ZEPHYR_BASE is unset.
```

Comment:
But we already have a default for `ZEPHYR_BASE`:

https://github.com/zephyrproject-rtos/zephyr/blob/ee9903005a6cebdce4c0cc5b8bf8144c9ec8ab7c/scripts/ci/check_compliance.py#L1110-L1118

Reply:
> But we already have a default for ZEPHYR_BASE:

Oh, I didn't see that this function was changing global variables. That's fun.

I still need this at module level so that `check_compliance: re-work DevicetreeBindingsCheck` later on in the PR works, so I'll just refactor this patch to remove the lines you mention as well.

Reply:
But you did not remove those lines, did you?